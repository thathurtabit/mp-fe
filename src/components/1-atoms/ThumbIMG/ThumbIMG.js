import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { TransitionGroup, CSSTransition } from 'react-transition-group';
import IMG from './ThumbIMG.styled';
import LoadingSmall from '../LoadingSmall/LoadingSmall';
import { loadDelay } from '../../../utils/constants/constants';

export default class ThumbIMG extends Component {
  constructor(props) {
    super(props);
    this.state = { loading: true };
  }

  componentDidMount() {
    // Only show images once load has complete
    // Note: Must be a prettier way of doing this
    const img = new Image();
    const { url } = this.props;
    img.src = url;
    img.onload = () => this.setState({ loading: false });
  }

  render() {
    const { loading } = this.state;
    const { title, url } = this.props;
    return (
      <Fragment>
        {loading && <LoadingSmall />}
        {!loading && (
          <TransitionGroup>
            <CSSTransition
              in={!loading}
              classNames="thumb"
              appear
              timeout={loadDelay}
            >
              <IMG className="thumb" src={url} alt={title} />
            </CSSTransition>
          </TransitionGroup>
        )}
      </Fragment>
    );
  }
}

ThumbIMG.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
