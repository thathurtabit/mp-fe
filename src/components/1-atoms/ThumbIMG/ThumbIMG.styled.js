import styled from 'styled-components';

const IMG = styled.img`
  border: 1px solid ${props => props.theme.links.hover};
  box-shadow: 5px 5px 0 rgba(0, 0, 0, 0.05);
  max-width: 250px;
  opacity: 0;
  transform: translateY(20px);
  transition: box-shadow ${props => props.theme.thumb.transition.duration}
      ${props => props.theme.transition.easeOut},
    transform ${props => props.theme.thumb.transition.duration}
      ${props => props.theme.transition.easeOut},
    opacity ${props => props.theme.thumb.transition.duration}
      ${props => props.theme.transition.easeOut};
  width: 100%;

  &:hover {
    box-shadow: 10px 10px 0 ${props => props.theme.colors.brand};
  }

  &.thumb-enter-done {
    opacity: 1;
    transform: translateY(0);
  }
`;

export default IMG;
