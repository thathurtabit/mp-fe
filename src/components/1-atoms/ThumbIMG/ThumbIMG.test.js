import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';
import { shallowWithTheme } from '../../../utils/test_config/testHelpers';

import ThumbIMG from './ThumbIMG';

describe('ThumbIMG', () => {
  it('should render correctly', () => {
    const element = shallowWithTheme(
      <ThumbIMG url="myURL" title="Test title" />
    );
    expect(shallowToJson(element)).toMatchSnapshot();
  });
});
