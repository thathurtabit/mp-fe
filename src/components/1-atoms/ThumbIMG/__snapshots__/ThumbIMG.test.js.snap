// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThumbIMG should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
